### DISCUSSION TITLE
Got your LibreSquares up and running

### DISCUSSION THREAD

**Discussion by nicvagn** [HarshilPatel007/YACS#4] [Show and tell]
I have gotten it up and running. Very impressive work mate. As it stands, with it I can import a pgn from my computer, and all the chapters show up. It lacks variations or lines I think, but this is a great jumping off point.

I will start porting it over to YACS

**Comment by HarshilPatel007** [comment]
Hi @nicvagn ,

That's great.
Please Copy gists code and create your new gist from it. (Don't fork) as I'm going to private it. (I don't want to embrace myself 😔)
And see my comment there.

Please update me after creating your gist So I can private it.

**Reply by HarshilPatel007** [reply]
> It lacks variations or lines I think, but this is a great jumping off point.

Yes, it lacks the support of variation/ sub-variations & comments.

I'm waiting for the python-chess PR to be merged. (It'll allow users to work with multiple comments)
But, to be frank, I had hard luck to code this.
As I was planning to make ( adding variation/sub-variations, add/ edit comments ) functions first and re use them for reading a PGN file. (For users to add to their game while playing)

So, planning was to create common functions for this workflow rather than creating specifically for reading only.

**Reply by nicvagn** [reply]
I have it locally, you can private

**Reply by HarshilPatel007** [reply]
@nicvagn ,
Feel free to ask questions/ doubts regarding the gist code here only.
So, It'll be easy to track and manage.

**Comment by nicvagn** [comment]
I downloaded the src, Is there and documentation I should grab?

**Reply by HarshilPatel007** [reply]
Sorry, there is no documentation for the code. I've added comments on a necessary part in the code.

(If you're talking about the qt, then just Google PySide6 document)

But, we should really think how we're going to manage the moves or pgn?

To show them in a table (as currently it is)
If yes, then how will we show the variants/sub-variations and comments? (Just like LucasChess)

Or to show just like how chessbase is doing.
